portal-api/src/v1/controllers/utilisation-report-service/utilisation-report-upload.controller.js

```diff
@@ -62,35 +64,74 @@ const sendEmailToBankPaymentOfficerTeam = async (reportPeriod, bankId, submitted
   }
 };
 
+/**
+ * Saves file to Azure in utilisation-reports ShareClient, returns the file storage info
+ * @param {object} file
+ * @param {string} bankId - bank id as a string
+ * @returns {Promise<{fileInfo: object, error: boolean}>} - if file is not saved error is true, otherwise returning
+ * azure storage details with folder & file name, full path & url.
+ */
+const saveFileToAzure = async (file, bankId) => {
+  try {
+    console.info(`Attempting to save utilisation report for bank: ${bankId}`);
+    const { originalname, buffer } = file;
+
+    const fileInfo = await uploadFile({
+      fileshare: FILESHARES.UTILISATION_REPORTS,
+      folder: bankId,
+      filename: formatFilenameForSharepoint(originalname),
+      buffer,
+      allowOverwrite: true,
+    });
+
+    console.info(`Successfully saved utilisation report for bank: ${bankId}`);
+    return { fileInfo, error: false };
+  } catch (error) {
+    console.error('Failed to save utilisation report: %O', error);
+    return { fileInfo: null, error: true };
```

Comment:
I think we should just throw here and let the error bubble up to the initial controller function where `res.status` is set. This prevents the need to remember to check the response for success. We then know that if the call to `saveFileToAzure` is successful the the response will be the populated `fileInfo`. So this function would then be
```
 /**
 * Saves file to Azure in utilisation-reports ShareClient, returns the file storage info
 * @param {object} file
 * @param {string} bankId - bank id as a string
 * @returns {Promise<object>} - azure storage details with folder & file name, full path & url.
 */
const saveFileToAzure = async (file, bankId) => {
  try {
    console.info(`Attempting to save utilisation report for bank: ${bankId}`);
    const { originalname, buffer } = file;

    const fileInfo = await uploadFile({
      fileshare: FILESHARES.UTILISATION_REPORTS,
      folder: bankId,
      filename: formatFilenameForSharepoint(originalname),
      buffer,
      allowOverwrite: true,
    });

    console.info(`Successfully saved utilisation report for bank: ${bankId}`);
    return fileInfo;
  } catch (error) {
    console.error('Failed to save utilisation report: %O', error);
    throw error;
  }
};
```
and when calling in `uploadReportAndSendNotification`, instead of
```
const { fileInfo, error } = await saveFileToAzure(file, parsedUser.bank.id);
if (error) {
  const status = 500;
  return res.status(status).send({ status, data: 'Failed to save utilisation report to Azure' });
}
```
we can just do
```
const fileInfo = await saveFileToAzure(file, parsedUser.bank.id);
```
we don't need to replace the `if (error) {` with a try/catch as the `uploadReportAndSendNotification` already has its own and handles the error with an appropriate response.